### DISCUSSION TITLE
How to use SNPE provider?

### DISCUSSION THREAD

**Discussion by VictoryDong** [microsoft/onnxruntime#13244] [Other Q&A]
There are too few documents for using SNPE, so I don't know how to load the model into a session, so far I have done a little experiment:
1. Load the onnx format model and it can run normally, but looking at the cpu ratio, it should be that the dsp is not used, and all run on the cpu.
2. When loading the dlc format model, an error is reported directly, and the model cannot be loaded correctly。

Can anyone help me？

more Information：
I followed the documentation steps to compile the library using snpe and configure the options of snpe。

```
std::unordered_map<std::string, std::string> snpe_options;
 snpe_options["runtime"] = "DSP";
 snpe_options["buffer_type"] = "FLOAT";
 sessionOptions->AppendExecutionProvider("SNPE", snpe_options);
```

**Comment by ytaous** [comment]
are you following this doc? https://onnxruntime.ai/docs/execution-providers/SNPE-ExecutionProvider.html
What's the error?

@HectorSVC

**Reply by ytaous** [reply]
answered here - https://github.com/microsoft/onnxruntime/issues/13247

**Comment by itskyf** [comment]
Is there any to when convert the ONNX model with embedded DLC to .ORT format?
